Hazel/Dtls/DtlsUnityConnection.cs

```diff
@@ -213,23 +225,34 @@ protected override void ResendPacketsIfNeeded()
                     DateTime now = DateTime.UtcNow;
                     if (now >= this.nextEpoch.NextPacketResendTime)
                     {
-                        switch (this.nextEpoch.State)
+                        double negotationDuration = (now - this.nextEpoch.NegotationStartTime).TotalMilliseconds;
+                        this.nextEpoch.PacketResendCount++;
+
+                        if ((this.ResendLimit > 0 && this.nextEpoch.PacketResendCount > this.ResendLimit)
+                            || negotationDuration > this.DisconnectTimeout)
                         {
-                            case HandshakeState.ExpectingServerHello:
-                            case HandshakeState.ExpectingCertificate:
-                            case HandshakeState.ExpectingServerKeyExchange:
-                            case HandshakeState.ExpectingServerHelloDone:
-                                this.SendClientHello();
-                                break;
+                            this.DisconnectInternal(HazelInternalErrors.DtlsNegotiationFailed, $"DTLS negotiation failed after {this.nextEpoch.PacketResendCount} resends ({(int)negotationDuration} ms).");
```

Comment:
Great! I think this solves one of the error cases that've been working on @nommiin 

We might want to expand on this a bit to detect a "server never reached" case where no packets were ever received, and I think we can work from this